### DISCUSSION TITLE
Make Livewire.on() Listener to work with call of $this->dispatch() from ALL Lifecycle Hooks

### DISCUSSION THREAD

**Discussion by Ronan-Lenor** [livewire/livewire#8773] [3. Feature ideas]
Ok, so i figured out that Livewire 3 can't Listen some events that are dispatched from some Lifecycle Hooks.
Only those 4 works : `hydrate`, `boot`, `rendering`, `rendered`.

Could the Livewire team make it work also for `updating`, `updated`, `dehydrate` ?

As nothing is mentioned about his in the Upgrade guide from v2 to v3, then i guess it could be also a potential bug ?

I also noticed that using `x-init="Livewire.on()"` don't listen dispatch() that come from cutom methods while `@script<script>Livewire.on()</script>@endscript` does.

can it be corrected ???

**Comment by nastoychev** [comment]
### Workaround:
1. Place dispatch in your Livewire Component file
  ```php
  
  public function updated(): void
    {
       $this->dispatch('notification:show', message: 'Debug message!');
    }
  
  ```
2. Place in your .blade.php file
  ```html
  @script
  <script>
      Livewire.on('notification:show', event => {
          console.log('Event received:', event.message);
      });
  </script>
  @endscript
  ```
This workaround allows you to properly listen for Livewire events in scenarios where `x-init="Livewire.on()"` doesn't work due to the limitations in Livewire 3.x. Using `@script` ensures that your event listeners are set up in a way that reliably catches events emitted from any Livewire lifecycle hook.

Livewire documentation:
- [https://livewire.laravel.com/docs/events#listening-for-events-inside-component-scripts](https://livewire.laravel.com/docs/events#listening-for-events-inside-component-scripts)
- [https://livewire.laravel.com/docs/javascript#interacting-with-events](https://livewire.laravel.com/docs/javascript#interacting-with-events)
